src/java.base/share/classes/jdk/crac/impl/OrderedContext.java

```diff
@@ -28,55 +28,43 @@
 
 import jdk.crac.*;
 
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.WeakHashMap;
+import java.util.*;
 
-public class OrderedContext<R extends Resource> extends AbstractContextImpl<R> {
-    private final String name;
-    private boolean checkpointing = false;
-    protected long order = 0;
-    protected final WeakHashMap<R, Long> resources = new WeakHashMap<>();
+/**
+ * Context performing Checkpoint notification in reverse order of registration.
+ * Concurrent registration along notification is silently ignored.
+ * @param <R>
+ */
+public class OrderedContext<R extends Resource> extends AbstractContext<R> {
```

Comment:
This class should not be used anywhere directly, and is not exposed to users => should be abstract.

Reply:
By it's own it's a valid Contex implementation.